src/app/sidebar-nav/sidebar-nav.component.ts

```diff
@@ -28,17 +28,20 @@ export class SidebarNavComponent implements OnInit, Input {
       error => this.errorMessage = error);
 
     this.route.queryParams.subscribe((params) => {
-      this.id = +params['id'];
+      this.id = params['id'];
       this.view = params['view'] ? params['view'] : 'chart';
       this.yoy = params['yoy'] ? params['yoy'] : 'false';
       this.ytd = params['ytd'] ? params['ytd'] : 'false';
       const search = params['search'];
-      if (this.id) {
-        this.selectedCategory = this.id;
-      } else if (isNaN(this.id)) {
-        this.selectedCategory = null;
-      } else {
+      if (!this.id || this.id === undefined) {
         this.selectedCategory = 42;
+      } else {
```

Comment:
This is a good place for the early return pattern: https://forum.freecodecamp.com/t/the-return-early-pattern/19364